app/src/main/java/mozilla/lockbox/adapter/SettingListAdapter.kt

```diff
@@ -87,6 +87,34 @@ class SettingListAdapter : RecyclerView.Adapter<SettingViewHolder>() {
                 holder.text = configuration.text
             }
         }
+        setContentDescription(holder)
+    }
+
+    private fun setContentDescription(holder: SettingViewHolder) {
+        when (holder) {
+            is TextSettingViewHolder -> {
+                holder.view.contentDescription = holder.view.resources.getString(R.string.auto_lock_description)
+            }
+            is AppVersionSettingViewHolder -> {
+                holder.view.contentDescription = holder.view.resources.getString(R.string.app_version_description)
+            }
+            is ToggleSettingViewHolder -> {
+                val title = holder.view.resources.getString(holder.title)
+                when (title) {
+                    holder.view.resources.getString(R.string.autofill) -> {
+                        holder.toggle.contentDescription =
+                            holder.view.resources.getString(R.string.autofill_description)
+                    }
+                    holder.view.resources.getString(R.string.unlock) -> {
+                        holder.toggle.contentDescription = holder.view.resources.getString(R.string.send_usage_data)
+                    }
+                    holder.view.resources.getString(R.string.send_usage_data) -> {
+                        holder.toggle.contentDescription =
+                            holder.view.resources.getString(R.string.fingerprint_description)
+                    }
+                }
+            }
+        }
```

Comment:
We should be aiming for a line in the method above (e.g. L75) such as: 

```kotlin
holder.contentDescription = configuration.contentDescription
```

On the whole, having a `switch`/`when` statement based on class is a signal that the code in the case blocks should go in the classes themselves.

So;

```kotlin
class TextSettingViewHolder { // whatever

   var contentDescription: Int = R.string.empty_string
        set(@StringRes value) {
            view.description.setContentDescription(value)
        }
}
```

As it stands, next time we add a new settings with a `TextSettingConfiguration`, we'll have the same `auto_lock_description`.

So we should have 

```kotlin
abstract class SettingCellConfiguration(
    @StringRes open val title: Int? = null,
    @StringRes open val subtitle: Int? = null
    @StringRes open val contentDescription: Int? = null
)

class TextSettingConfiguration(
    @StringRes title: Int,
    @StringRes subtitle: Int? = null,
    @StringRes contentDescription: Int,
    val detailTextDriver: Observable<Int>,
    val clickListener: Consumer<Unit>
) : SettingCellConfiguration(title, subtitle, contentDescription)
```

This strategy will tidy up the `when` within a `when` you have here.

Reply:
Understood. For context, the reason that I added the separate method was that we may have more than one content description for multiple items within the setting based on what we find with TalkBack/Accessibility Scanner/etc :) 

That being said, I agree with your proposed approach and I am going to work on those changes now :D